packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.ts

```diff
@@ -248,16 +227,6 @@ export class AdapterMomentHijri extends AdapterMoment implements MuiPickersAdapt
       throw new Error('max date must be on or before 1499-12-29 H (2076-11-26)');
     }
 
-    const startDate = this.moment(start).startOf('iYear');
-    const endDate = this.moment(end).endOf('iYear');
-    const years: Moment[] = [];
-
-    let current = startDate;
-    while (current.isBefore(endDate)) {
-      years.push(current);
-      current = current.clone().add(1, 'iYear');
-    }
-
-    return years;
+    return super.getYearRange([start, end]);
```

Comment:
This adapter is the only one to have a slightly different behavior on `getYearRange` to be able to throw errors when using invalid dates.

If we remove these errors, we can remove `getYearRange` from the adapter and just add a method on the pickers utils.
But the error probably have some value.